packages/core/src/controllers/subscriber.ts

```diff
@@ -450,46 +492,49 @@ export class Subscriber extends ISubscriber {
   private async batchSubscribe(subscriptions: SubscriberTypes.Params[]) {
     if (!subscriptions.length) return;
 
-    const result = (await this.rpcBatchSubscribe(subscriptions)) as string[];
-    if (!isValidArray(result)) return;
-
-    this.onBatchSubscribe(result.map((id, i) => ({ ...subscriptions[i], id })));
+    await this.rpcBatchSubscribe(subscriptions);
+    this.onBatchSubscribe(
+      subscriptions.map((s) => ({ ...s, id: this.getSubscriptionId(s.topic) })),
+    );
   }
 
+  // @ts-ignore
   private async batchFetchMessages(subscriptions: SubscriberTypes.Params[]) {
     if (!subscriptions.length) return;
     this.logger.trace(`Fetching batch messages for ${subscriptions.length} subscriptions`);
     const response = await this.rpcBatchFetchMessages(subscriptions);
     if (response && response.messages) {
-      this.pendingBatchMessages = this.pendingBatchMessages.concat(response.messages);
+      await new Promise<void>((resolve) => {
+        setTimeout(async () => {
+          await this.relayer.handleBatchMessageEvents(response.messages);
+          resolve();
+        }, 1000);
+      });
```

Comment:
I feel like this could be way more concise and readable
```suggestion
await delay(1000); // or sleep(1000)
await this.relayer.handleBatchMessageEvents(response.messages);
```

Implementation of the sleep or delay fn
```ts
function delay(ms: number) {
  return new Promise<void>((resolve) => setTimeout(resolve, ms));
}
```